code/game/objects/structures/girders.dm

```diff
@@ -176,6 +176,33 @@
 		if(STATE_REINFORCED_WALL)
 			return do_reinforced_wall(W, user)
 		if(STATE_DISPLACED)
+			if(HAS_TRAIT(W, TRAIT_TOOL_BLOWTORCH))
+				var/area/area = get_area(src)
+				if(CEILING_IS_PROTECTED(area.ceiling, CEILING_GLASS))
+					to_chat(user, SPAN_WARNING("Watchtowers can only be built in the open."))
+					return
+					
+				var/list/turf/turfs = CORNER_BLOCK(get_turf(src), 2, 2)
+				var/list/obj/structure/girder/girders = list()
+
+				for(var/turf/current_turf in turfs)
+					var/found_girder = FALSE
+					for(var/obj/structure/girder/girder in current_turf)
+						if(girder.state == STATE_DISPLACED)
+							found_girder = TRUE
+							girders += girder
+					if(!found_girder)
+						return
+
+
+				if(!do_after(user, 3 SECONDS, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+					return
+
+				new /obj/structure/watchtower(loc)
+				
+				for(var/list/obj/structure/girder as anything in girders)
+					qdel(girder)
```

Comment:
There is zero to_chat feedback on this action both before or after the do_after. Nor any sound e.g. `playsound(loc, 'sound/items/Welder2.ogg', 25, 1)`